### DISCUSSION TITLE
Can I use both oh-my-zsh and zim together?

### DISCUSSION THREAD

**Discussion by NamekMaster** [zimfw/zimfw#472] [Q&A] [answered]
I'd like to keep using oh-my-zsh, and use zim to install other plugins. Are their any potential problems?

**Comment by ericbn** [comment, marked as answer]
Yes, you can use both together. There are basically two ways to do it:
1. Install both ohmyzsh and Zim (you'll have both `source $ZSH/oh-my-zsh.sh` and `source ${ZIM_HOME}/init.zsh` in your ~/.zshrc file). In this case you should avoid using the Zim completion module, as it will conflict with ohmyzsh. Other Zim modules can conflict with ohmyzsh plugins too. Then you use zimfw to install other plugins. I don't recommend this approach.
2. Only install Zim, and not run the ohmyzsh installation (you'll have `source ${ZIM_HOME}/init.zsh` in your ~/.zshrc file, but not `source $ZSH/oh-my-zsh.sh`). Then use zimfw to install the ohmyzsh plugins you want, and any other plugins you also want. I strongly suggest this approach.

With zimfw version 1.10.0, which was just released, the second approach got easier to manage. With the new `--root` option in `zmodule`, you can add multiple `zmodule` calls for all the multiple ohmyzsh plugins you might want use. Example:
```
zmodule ohmyzsh/ohmyzsh --root plugins/perl
zmodule ohmyzsh/ohmyzsh --root plugins/vim-interaction
```

**Comment by LittleNewton** [comment]
Hi @ericbn,

I am using oh-my-zsh and Zim at the same time. I have found that I have to init zim and then init oh-my-zsh. If the order can not be satisfied, the system will have trouble.

Nowadays, what I need from oh-my-zsh is `agnoster` theme. So is there a way to use ohmyzsh's theme without install ohmyzsh whole?

Thanks.

**Reply by ericbn** [reply]
We have an [eriner](https://github.com/zimfw/eriner) prompt theme that is very similar to agnoster. Make sure you have this in your ~/.zimrc to enable it:
```
zmodule git-info
zmodule prompt-pwd
zmodule eriner
```
Also make sure you don't have other themes enabled (mainly after eriner) in ~/.zimrc, like our default asciiship theme for example, otherwise they'll override eriner.

If you want the eriner theme to look exactly like agnoster, add this to your ~/.zshrc (not the ~/.zimrc):
```
PWD_COLOR=blue
```
Check the [eriner](https://github.com/zimfw/eriner) documentation for other customization settings.

**Reply by LittleNewton** [reply]
eriner theme is also good to use. However, its prompt-pwd is is compact mode. For example, if my current working directory is 

```
/mnt/Intel_750_RAID-Z1/Documents/git_repo/container_remake
```

eriner theme will show it like

```
/m/I/D/g/container_remake
```

I didn't find the solution from eriner's documentation. Is there a method to turn the compact mode to normal mode?

**Reply by ericbn** [reply]
Under [Advanced settings](https://github.com/zimfw/eriner/blob/master/README.md#advanced-settings) you'll find a link to the [prompt-pwd module settings](https://github.com/zimfw/prompt-pwd/blob/master/README.md#settings).

TL;DR Add this to the end of your ~/.zshrc file
```
zstyle ':zim:prompt-pwd:fish-style' dir-length 0
```

**Reply by LittleNewton** [reply]
Thanks for your help! It works well.

Before I thought the settings would be in `eriner`, but actually, it is located in `prompt-pwd`. 

:)
